deanonymization/check_exif.go

```diff
@@ -2,45 +2,40 @@ package deanonymization
 
 import (
 	"bytes"
+	"github.com/rwcarlsen/goexif/exif"
+	"github.com/rwcarlsen/goexif/tiff"
 	"github.com/s-rah/onionscan/config"
 	"github.com/s-rah/onionscan/report"
-	"github.com/xiam/exif"
-	"io"
 	"net/url"
 	"strings"
 )
 
-func CheckExif(osreport *report.OnionScanReport, anonreport *report.AnonymityReport, osc *config.OnionScanConfig) {
+type ExifWalker struct {
+	osreport   *report.OnionScanReport
+	anonreport *report.AnonymityReport
+	osc        *config.OnionScanConfig
+}
+
+func (w *ExifWalker) Walk(name exif.FieldName, val *tiff.Tag) error {
+	w.anonreport.AddExifTag(string(name), val.String())
+	return nil
+}
 
+func CheckExif(osreport *report.OnionScanReport, anonreport *report.AnonymityReport, osc *config.OnionScanConfig) {
 	for _, id := range osreport.Crawls {
 		crawlRecord, _ := osc.Database.GetCrawlRecord(id)
 
 		if crawlRecord.Page.Status == 200 && strings.Contains(crawlRecord.Page.Headers.Get("Content-Type"), "image/jpeg") {
-			reader := exif.New()
-			_, err := io.Copy(reader, bytes.NewReader(crawlRecord.Page.Raw))
-
-			// exif.FoundExifInData is a signal that the EXIF parser has all it needs,
-			// it doesn't need to be given the whole image.
-			if err != nil && err != exif.ErrFoundExifInData {
-				// We don't care if we fail
-				return
-			}
-
-			err = reader.Parse()
+			exifdata, err := exif.Decode(bytes.NewReader(crawlRecord.Page.Raw))
 
 			if err != nil {
 				// We don't care if we fail
 				return
```

Comment:
Eh, good catch, skipping all further images wasn't the intention here, will change.